packages/cli/src/server/api/memory/rooms.ts

```diff
@@ -0,0 +1,170 @@
+        type: type,
+        channelId: roomId,
+        serverId: serverId,
+        worldId: resolvedWorldId,
+        metadata: metadata,
+      });
+
+      await runtime.addParticipant(runtime.agentId, roomId);
+      await runtime.ensureParticipantInRoom(runtime.agentId, roomId);
+      await runtime.setParticipantUserState(roomId, runtime.agentId, 'FOLLOWED');
+
+      sendSuccess(
+        res,
+        {
+          id: roomId,
+          name: name,
+          agentId: agentId,
+          createdAt: Date.now(),
+          source: source,
+          type: type,
+          worldId: resolvedWorldId,
+          serverId: serverId,
+          metadata: metadata,
+        },
+        201
+      );
+    } catch (error) {
+      logger.error(`[ROOM CREATE] Error creating room for agent ${agentId}:`, error);
+      sendError(res, 500, 'CREATE_ERROR', 'Failed to create room', error.message);
+    }
+  });
+
+  // Get all rooms where an agent is a participant
+  router.get('/:agentId/rooms', async (req, res) => {
+    const agentId = validateUuid(req.params.agentId);
+    if (!agentId) {
+      return sendError(res, 400, 'INVALID_ID', 'Invalid agent ID format');
+    }
+
+    const runtime = agents.get(agentId);
+    if (!runtime) {
+      return sendError(res, 404, 'NOT_FOUND', 'Agent not found');
+    }
+
+    try {
+      const worlds = await runtime.getAllWorlds();
+      const participantRoomIds = await runtime.getRoomsForParticipant(agentId);
+      const agentRooms = [];
+
+      for (const world of worlds) {
+        const worldRooms = await runtime.getRooms(world.id);
+        for (const room of worldRooms) {
+          if (participantRoomIds.includes(room.id)) {
+            agentRooms.push({
+              ...room,
+              worldName: world.name,
+            });
+          }
+        }
+      }
+
+      sendSuccess(res, { rooms: agentRooms });
+    } catch (error) {
+      logger.error(`[ROOMS LIST] Error retrieving rooms for agent ${agentId}:`, error);
+      sendError(res, 500, 'RETRIEVAL_ERROR', 'Failed to retrieve agent rooms', error.message);
+    }
+  });
+
+  // Get room details
+  router.get('/:agentId/rooms/:roomId', async (req: CustomRequest, res: express.Response) => {
+    const agentId = validateUuid(req.params.agentId);
+    const roomId = validateUuid(req.params.roomId);
+
+    if (!agentId || !roomId) {
+      return sendError(res, 400, 'INVALID_ID', 'Invalid agent ID or room ID format');
+    }
+
+    // Get runtime
+    const runtime = agents.get(agentId);
+    if (!runtime) {
+      return sendError(res, 404, 'NOT_FOUND', 'Agent not found');
+    }
+
+    try {
+      const room = await runtime.getRoom(roomId);
+      if (!room) {
+        return sendError(res, 404, 'NOT_FOUND', 'Room not found');
+      }
+
+      // Optionally, enrich room data with world name if needed
+      // This depends on whether getRoom returns worldId and if you want to include worldName
+      let worldName = 'N/A';
```

Comment:
seems wrong place to assign worldName N/A. Should happen on world creation and be defaulted rather then in the room GET.